testsuite/ui/widgets/__init__.py

```diff
@@ -17,52 +17,91 @@
 # pylint: disable=abstract-method
 class RadioGroup(GenericLocatorWidget):
     """
-    Radio group of 3scale pages. It contains both radio and check box elements, therefore it is not
-    typical radio element. No control of option switching is implemented.
-
-    In case of check box (like one on the bottom of the User Edit page), user needs to remember that
-    possibility of multiple selected options is allowed (not an typical radio behaviour)
+    Radio group of 3scale pages.
     """
-    OPTIONS_SECTION = './fieldset/ol[contains(@class, "{}")]'
+    OPTIONS_SECTION_CLASS = './fieldset/ol[contains(@class, "{}")]'
+    OPTIONS_SECTION = './fieldset/ol'
 
     OPTIONS = './li'
-    OPTIONS_BY_ID = OPTIONS + '/label/input[@id="{}"]'
+    OPTIONS_INPUT = OPTIONS + '/label/input'
+    OPTIONS_BY_ID = OPTIONS_INPUT + '[@id="{}"]'
 
     def __init__(self, parent=None, locator=None, fieldset_id="", logger=None):
         super().__init__(parent, locator, logger)
         if fieldset_id:
-            self.options_section = self.browser.element(self.OPTIONS_SECTION.format(fieldset_id))
+            self.options_section = self.browser.element(self.OPTIONS_SECTION_CLASS.format(fieldset_id))
         else:
-            self.options_section = self.browser.element(locator)
+            self.options_section = self.browser.element(self.OPTIONS_SECTION)
 
-    def select(self, options):
+    def select(self, option: str):
         """
         Select radio (check box) element from the list.
-        :param options: String id-s of options
+        :param option: String id-s of options
+        """
+        element = self.browser.element(self.OPTIONS_BY_ID.format(option), parent=self.options_section)
+        element.click()
+
+
+# pylint: disable=abstract-method
+# Widget contains fill method which raise not implemented exception if widget is not fillable but pylint detect it as
+# an abstract method
+class CheckBoxGroup(GenericLocatorWidget):
+    """
+    CheckBox group of 3scale pages
+    :param ol_class: Set if Checkbox group identifier is in ol level of group.
+                     e.g. fieldse/ol[class=ServiceList]
+    :param field_set_name: Set if Checkbox group identifier is in fieldset level of group.
+    """
+    OPTIONS_SECTION = './fieldset/ol'
+    OPTIONS_SECTION_OL_CLASS = './/fieldset/ol[contains(@class, "{}")]'
+    OPTIONS_SECTION_FIELDSET_NAME = './/fieldset[contains(@name, "{}")]/ol'
+
+    OPTIONS = './li'
+    OPTIONS_INPUT = OPTIONS + '/label/input'
+    OPTIONS_BY_ID = OPTIONS_INPUT + '[@id="{}"]'
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, parent=None, locator=None, ol_class=None, field_set_name=None, logger=None):
+        super().__init__(parent, locator, logger)
+        if ol_class:
+            self.options_section = self.browser.element(self.OPTIONS_SECTION_OL_CLASS.format(ol_class))
```

Comment:
`ol_class` and `field_set_name` should be saved in locator of the widget.. At least that is common usage in widgetastic

Reply:
If you save it into locator you save some code. Example from your code:
`self.browser.element(self.OPTIONS_BY_ID.format(option), parent=self.options_section)`
with locator:
`self.browser.element(self.OPTIONS_BY_ID.format(option))`

Reply:
Changed to locator form